src/services/submissionService.ts

```diff
@@ -301,35 +301,40 @@ export const approveSubmission = async (submissionId: string): Promise<Result<gq
 
             return approveResult
         }
-
-        //update the submission to approved
-        currentSubmission.isApproved = true
-        currentSubmission.isUnderReview = false
-        currentSubmission.updatedDate = new Date().toISOString()
-
-        await submissionRef.set(currentSubmission, { merge: true })
-
+    
         logger.info(`\nDB-UPDATE: Submission ${submissionId} was approved.`)
-
+        
+        //try creating the facility
+        const createFacilityResult = await createFacility(currentSubmission.facility as dbSchema.Facility)
+        
+        if (createFacilityResult.hasErrors) {
+            approveResult.errors?.concat(createFacilityResult.errors!)
+            return approveResult
+        }
+        
         //try creating healthcare professional(s)
         for await (const healthcareProfessional of currentSubmission.healthcareProfessionals ?? []) {
             const healthcareProfessionalInput
-                = healthcareProfessional satisfies gqlTypes.CreateHealthcareProfessionalInput
+            = healthcareProfessional satisfies gqlTypes.CreateHealthcareProfessionalInput
+            
+            healthcareProfessionalInput.facilityIds.push(createFacilityResult.data.id)
             const createHealthcareProfessionalResult = await createHealthcareProfessional(healthcareProfessionalInput)
-
+            
             if (createHealthcareProfessionalResult.hasErrors) {
                 approveResult.errors?.concat(createHealthcareProfessionalResult.errors!)
                 return approveResult
             }
         }
 
-        //try creating the facility
-        const createFacilityResult = await createFacility(currentSubmission.facility as dbSchema.Facility)
+        //update the submission to approved only if creating the facility and healthcare professionals succeeds
+        currentSubmission.isApproved = true
+        currentSubmission.isUnderReview = false
+        currentSubmission.updatedDate = new Date().toISOString()
 
-        if (createFacilityResult.hasErrors) {
-            approveResult.errors?.concat(createFacilityResult.errors!)
-            return approveResult
-        }
+        await submissionRef.set(currentSubmission, { merge: true })
+
+        //set the data to return to the graphQl response
```

Comment:
nit: this function might not be called from a resolver, so we shouldn't imply it's a graphql response